### DISCUSSION TITLE
Small common project

### DISCUSSION THREAD

**Discussion by t-redd** [imk-toolkit/imk-toolkit#76] [Ideas]
I would propose a small project to get a feeling for the environment:
in extension to the num_den method, we could write a partial column routine. I have set up a kind of wish list as arg list, which may implemented little by little.

def main( data_array, p_mid=None, p_low=None, p_top=None, unit='mol/m2', cumcol=False, down=True ):
"""Calculates partial columns of a trace gas in an atmosphere from its mixing ratio

The partial column of a cell layer is calculated. Within the cell the mixing ratio is assumed to be constant. Pressure can be given at cell centres or plus at top and bottom boundaries. In first case, cell boundaries are calculated as the corresponding pressure of the mean of neighboured log pressure. Optionally, the cumulative partial columns can be returned. Extensions to thermosphere and other planets possible.
Args: 
    data_array: mixing ratio at cell centres
    temperature: temp at cell centres, same shape as data_array 
    p_mid: pressure in cell centers, shape can be 1-D (fixed p-grid) or must be of shape of data_array
    p_low: pressure at cell bottom, see above
    p_top: pressure at celltop, see above
    unit: SI unit is mol/m2. Also possible: 'DU', '1/m2'
    H: scale height used to transform to pressure height
    cumcol: return cumulative column to cell centre instead of partial column of cell
    down: if cumcol calculate from top of atmosphere, if not from surface

Returns:
    A data array of same shape as input array with partial columns of cells or total columns
"""

**Comment by ucyo** [comment]
Nice! Lets just do it! I added a template file to the main script folders e.g. dataarray 

https://github.com/imk-toolkit/imk-toolkit/blob/94181eab27536b9364a1e078b4c51ecba3b7b6cb/imktk/dataarray_methods/_template.py#L1-L40
